docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -83,15 +83,15 @@
    private static final String NO_IMAGE = "jclouds-placeholder-for-image";
 
    protected List<String> dns = ImmutableList.of();
-   protected String hostname;
-   protected Integer memory;
-   protected Integer cpuShares;
-   protected List<String> entrypoint = ImmutableList.of();
-   protected List<String> commands = ImmutableList.of();
+   @Nullable protected String hostname;
+   @Nullable protected Integer memory;
+   @Nullable protected Integer cpuShares;
+   @Nullable List<String> entrypoint;
+   @Nullable List<String> commands;
```

Comment:
Which is the purpose of making them nullable then? Having null collections is a bad pattern and a potential source of NPEs; we should try to avoid them.